testdriver/testdriver.go

```diff
@@ -744,6 +744,7 @@ func Condition(cond string) (bool, error) {
 	switch f {
 	case govim.FlavorVim, govim.FlavorGvim:
 		cmd := cmd.BuildCommand("-v", "--version")
+		cmd.Env = append(os.Environ(), "LANGUAGE=en") // hardcode language to simplify parsing output
```

Comment:
We could run:
`cmd := cmd.BuildCommand("--cmd", "echo v:versionlong | qall", "--not-a-term")`
instead to avoid the need of parsing the huge vim version output.

The parseVimVersion could then be reduced to something like:
```
func parseVimVersion(b []byte) (string, error) {
    if len(b) < 7 {
        return "", fmt.Errorf("failed to calculate valid Vim version from %q", b)
    }
    return fmt.Sprintf("v%d.%d.%d", b[0], b[1:2], b[3:]), nil
}
```

And that would make the version_test.go & testdata files obsolete.

Reply:
Adapted a version of this that I got working with both `gvim` and `vim`. No idea why it didn't work earlier. Thanks for prompting me to check again